esrally/chart_generator.py

```diff
@@ -1371,14 +1371,19 @@ def index(environment, race_configs, title):
 
 
 class RaceConfigTrack:
-    def __init__(self, cfg, name=None):
-        self.cached_track = self.load_track(cfg, name)
+    def __init__(self, cfg, repository, name=None):
+        self.repository = repository
+        self.cached_track = self.load_track(cfg, name=name)
 
     def load_track(self, cfg, name=None, params=None):
         if not params:
             params = {}
+        # required in case a previous track using a different repository has specified the revision
+        if cfg.opts("track", "repository.name") != self.repository:
+            cfg.add(config.Scope.applicationOverride, "track", "repository.revision", None)
```

Comment:
(Unrelated to this PR) I wonder whether we should disallow specifying a track revision for the generator via a command line parameter.

Reply:
Yes I think this would make sense given that we allow mixing tracks from different repositories.